### PR TITLE
Docs: Move Maintainers to Committers

### PR DESCRIPTION
**Pull request by xmulligan**
Maintainers are being tracked in https://github.com/cilium/community/blob/main/roles/Maintainers.md

### CODE REVIEW

File: MAINTAINERS.md
Comment (aditighag) [inline MAINTAINERS.md]: Nyc, but since you are updating the file, you might as well update this.

```suggestion
by the "tophat role" which is rotated between all committers.
```

File: MAINTAINERS.md
Comment (aanm) [inline MAINTAINERS.md]:
```suggestion
by the "tophat role" which is rotated between all committers as well as by any committer is the tests were successful and the reviews from all codeowners were accepted.
```

Comment (xmulligan) [reply MAINTAINERS.md]: Thinking about this again, should we just end this sentence after CODEOWNERS since who merges could change again in the future?

Comment (joestringer) [reply MAINTAINERS.md]: I agree with Bill, there's no need to go into details about the mechanics of merging PRs in `MAINTAINERS.md`.

Comment (joestringer) [reply MAINTAINERS.md]: I even wonder if this paragraph around "Regardless of ..." should just be dropped entirely. That was there to make it clear that Maintainers are not elevated above the standards required for all Committers. The governance doc already clarifies this.

If we think that people will come to the manitainers doc to look for information about how to get PRs merged, then maybe a simple link to https://docs.cilium.io/en/stable/contributing/development/contributing_guide would suffice. But I'm also OK to just omit that.

Comment (xmulligan) [reply MAINTAINERS.md]: I'm going to delete it

**Review by joestringer — APPROVED** [review]

**Review by aanm — APPROVED** [review]

**Pull request merged**
